libbeat/tests/integration/framework.go

```diff
@@ -651,7 +665,17 @@ func createTempDir(t *testing.T) string {
 	cleanup := func() {
 		if !t.Failed() {
 			if err := os.RemoveAll(tempDir); err != nil {
-				t.Errorf("could not remove temp dir '%s': %s", tempDir, err)
+				// Ungly workaround Windows limitations
+				// Windows does not support the Interrup signal, so it might
+				// happen that Filebeat is still running, keeping it's registry
+				// file open, thus preventing the temporary folder from being
+				// removed. So we log the error and move on without failing the
+				// test
+				if runtime.GOOS == "windows" {
+					t.Logf("[WARN] Could not remove temporatry directory '%s': %s", tempDir, err)
+				} else {
+					t.Errorf("could not remove temp dir '%s': %s", tempDir, err)
+				}
```

Comment:
Is it happening that often? I'm wondering if it'd be worth retrying within a given timeout and/or perhaps getting a callback that should try to kill the beat once again. For any test that does not want/need/can try to stop the beat again, the callback can be a noop. That way this function does not depend on the `BeatProc` but still can attempt to stop it and retry the clean up.
It could receive the `BeatProc` if it wouldn't compromise other tests

Reply:
> Is it happening that often?

Often enough. While testing this PR on Windows, every time I added an `runtime.GOOS == "windows"` to fix a failure, another would pop up. They all together make the test fail all the time.

I'm a bit confused on what would be the function of the callback you're suggesting. One of the ideas of this integration test framework is to remove as much work from the engineer writing the test as possible, the clean up part being one of the key automation. So it feels that adding a callback for the clean up would be adding burden on the engineer writing tests.

Reply:
What if we do a few retries with an interval?

I don't think it's that much important though, the path leads to a unique temporary directory anyway, this should not conflict with following test runs.

Reply:
I'm afraid a few retries could lead to flakiness, because the path is a temporary folder, it won't be persisted for long on CI or on our machines when running the tests. I rather keep it simple and safe.